pkg/controller/apischeme/apischeme_controller.go

```diff
@@ -118,9 +123,73 @@ func (r *ReconcileAPIScheme) Reconcile(request reconcile.Request) (reconcile.Res
 		return reconcile.Result{}, nil
 	}
 
+	if cloudClient == nil {
+		cloudPlatform, err := utils.GetPlatformType(r.client)
+		if err != nil {
+			SetAPISchemeStatus(instance, "Couldn't reconcile", "Couldn't create a Cloud Client", cloudingressv1alpha1.ConditionError)
+			r.client.Status().Update(context.TODO(), instance)
+			return reconcile.Result{}, err
+		}
+		cloudClient = cloudclient.GetClientFor(r.client, *cloudPlatform)
+	}
+
+	serviceNamespacedName := types.NamespacedName{
+		Name:      instance.Spec.ManagementAPIServerIngress.DNSName,
+		Namespace: "openshift-kube-apiserver",
+	}
+
+	// Check for a deletion timestamp.
+	if instance.DeletionTimestamp.IsZero() {
+		// Request object is alive, so ensure it has the DNS finalizer.
+		if !controllerutil.ContainsFinalizer(instance, reconcileFinalizerDNS) {
+			controllerutil.AddFinalizer(instance, reconcileFinalizerDNS)
+			if err = r.client.Update(context.TODO(), instance); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
```

Comment:
Upon success, should this cause the object to requeue the Reconcile again, since it has been updated? If so, be sure to adjust the comment, and requeue.

Reply:
Is it necessary to requeue immediately after an update?  (I honestly don't know.)

Reply:
@mbarnes Yes. The reason is: When you write the object back to the API server, Kubernetes updates the version in the metadata and further writes to the original object will fail because the "source" object is older than what the API server knows about. This is to protect against overwriting, or time travelling to before previous saves.

General best practice is to do the write and return out of the Reconcile method to requeue again.

Reply:
That makes a lot of sense.  Thanks for the detailed explanation!

Reply:
On re-reading the logic, it returns anyway right after the update, so there's no subsequent updates.  The next reconciliation pass _should_ be a no-op, but I requeued anyway.  I guess it can't hurt.